source/contacts/RosterController.cpp

```diff
@@ -175,7 +176,7 @@ bool RosterController::updateAvailabilityForJid(const Swift::JID &jid, const Ros
     bool somethingChanged = false;
 
     QString localBareJid = QString::fromStdString(jid.toBare().toString());
-    appendToRosterIfNotAlreadyIn(localBareJid);
+    //appendToRosterIfNotAlreadyIn(localBareJid);
```

Comment:
Why do you have commented out all the 'appenToRoster...' calls?

Reply:
Because when leaving a group, an availability message can be received after and so it will restore the group in the list of contacts.